lib/ansible/module_utils/postgres.py

```diff
@@ -63,7 +66,38 @@ def postgres_common_argument_spec():
     )
 
 
-def connect_to_db(module, kw, autocommit=False):
+def connect_to_db(module, autocommit=False, fail_on_conn=True):
+
+    ensure_libs(module)
+
+    # To use defaults values, keyword arguments must be absent, so
+    # check which values are empty and don't include in the **kw
+    # dictionary
+    params_map = {
+        "login_host": "host",
+        "login_user": "user",
+        "login_password": "password",
+        "port": "port",
+        "ssl_mode": "sslmode",
+        "ca_cert": "sslrootcert"
+    }
+
+    # Might be different in the modules:
+    if module.params.get('db'):
+        params_map['db'] = 'database'
+    elif module.params.get('database'):
+        params_map['database'] = 'database'
+    elif module.params.get('login_db'):
+        params_map['login_db'] = 'database'
+
```

Comment:
Do we need a default case for this, e.g what happens if there is no `db`, `database` or `login_db` set, should we fail? Something like

```suggestion
        else:
            module.fail_json(msg="No database parameter set, need one of 'db', 'database' or 'login_db'")

```

Reply:
@andytom , nps, thank you much for reviewing!
Very interesting point about the default value.
However db/database/etc param is not necessary to be not None/'' for many of the modules, like postgresql_user, _idx, etc.
As we know in this case psycopg will use default db like a name of a user who's trying to connect.
Your suggestion gave me an idea to change it to module.warn, see my last commit.
I added warn_db_default arg for modules that work with global objects (user, settings, info).
I think it will be very useful if a dba forget to pass db name and we warn him about that he's working with a default database that is usually quite strange ;)
What do you think about it ?

Reply:
That makes perfect sense to me.

Reply:
@andytom , thank you for reviewing! Non having another channel to communicate with you, I sent a message to you in LinkedIn (if you like you can answer by mail to aaklychkov at mail dot ru, it I can't check LinkedIn before 20.00 UTC)